app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,10 @@
+class Admin::BaseController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_admin!
+
+  private
+
+  def require_admin!
+    redirect_to root_path unless current_user&.admin?
+  end
```

Comment:
Opiniones de llamarlo  `authenticate_admin!` y no añadir `authenticate_user!`?

```suggestion
  def authenticate_admin!
    redirect_to root_path unless current_user.admin?
  end
```

Reply:
1. Si no añadimos el `authenticate_user!` de devise no nos va redirigir a loguarnos
2. Llamarlo `authenticate_admin!` no me parece. El método no esta autenticando nada. En todo caso `authorize_admin!` pero prefiero `require`.